src/Compatibility/ApiCompat/Microsoft.DotNet.ApiCompatibility/Rules/CannotChangeGenericConstraints.cs

```diff
@@ -0,0 +1,169 @@
+            MetadataInformation leftMetadata,
+            MetadataInformation rightMetadata,
+            IList<CompatDifference> differences)
+        {
+            if (left is not IMethodSymbol leftMethod || right is not IMethodSymbol rightMethod)
+            {
+                return;
+            }
+
+            var leftTypeParameters = leftMethod.TypeParameters;
+            var rightTypeParameters = rightMethod.TypeParameters;
+
+            bool permitConstraintRemoval = !_settings.StrictMode && !leftMethod.IsVirtual;
+
+            CompareTypeParameters(leftTypeParameters, rightTypeParameters, left, permitConstraintRemoval, leftMetadata, rightMetadata, differences);
+        }
+
+        private void CompareTypeParameters(
+            ImmutableArray<ITypeParameterSymbol> leftTypeParameters,
+            ImmutableArray<ITypeParameterSymbol> rightTypeParameters,
+            ISymbol left,
+            bool permitConstraintRemoval,
+            MetadataInformation leftMetadata,
+            MetadataInformation rightMetadata,
+            IList<CompatDifference> differences)
+        {
+
+            Debug.Assert(leftTypeParameters.Length == rightTypeParameters.Length);
+            for (int i = 0; i < leftTypeParameters.Length; i++)
+            {
+                ITypeParameterSymbol leftTypeParam = leftTypeParameters[i];
+                ITypeParameterSymbol rightTypeParam = rightTypeParameters[i];
+
+                // Currently our symbol formatting for comparison includes generic parameter names rather than just arity
+                // https://github.com/dotnet/roslyn/issues/73051
+                // As a result this will only be called with matching parameters.
+                // When parameters don't match the symbol will be treated as different because it's key repesentation will differ.
+                // See SymbolEqualityComparer.GetKey https://github.com/dotnet/sdk/blob/5c8c0a75bba35c1fd0db879eb821c8635c142f79/src/Compatibility/ApiCompat/Microsoft.DotNet.ApiCompatibility/Comparing/SymbolEqualityComparer.cs#L22-L41
+                Debug.Assert(leftTypeParam.Name.Equals(rightTypeParam.Name));
+
+                List<string> addedConstraints = new();
+                List<string> removedConstraints = new();
+
+                CompareBoolConstraint(typeParam => typeParam.HasConstructorConstraint, "new()");
+                CompareBoolConstraint(typeParam => typeParam.HasNotNullConstraint, "notnull");
+                CompareBoolConstraint(typeParam => typeParam.HasReferenceTypeConstraint, "class");
+                CompareBoolConstraint(typeParam => typeParam.HasUnmanagedTypeConstraint, "unmanaged");
+                // unmanaged implies struct
+                CompareBoolConstraint(typeParam => typeParam.HasValueTypeConstraint & !typeParam.HasUnmanagedTypeConstraint, "struct");
+                // Not available
+                // CompareBoolConstraint(typeParam => typeParam.HasDefaultConstraint, "default");
```

Comment:
```suggestion
```

The `default` constraint is not part of the observable surface area of an API.  It's only used when disambiguating between overloads or in explicit interfaces.